pod/video/views.py

```diff
@@ -513,30 +517,68 @@ def my_videos(request):
         data_context["categories"] = cats
         data_context["videos_without_cat"] = videos_without_cat
 
+    videos_list = get_filtered_videos_list(request, videos_list)
+    videos_list = sort_videos_list(request, videos_list)
+    count_videos = len(videos_list)
+
     paginator = Paginator(videos_list, 12)
-    try:
-        videos = paginator.page(page)
-    except PageNotAnInteger:
-        videos = paginator.page(1)
-    except EmptyPage:
-        videos = paginator.page(paginator.num_pages)
+    videos = get_paginated_videos(paginator, page)
 
     if request.is_ajax():
         return render(
             request,
             "videos/video_list.html",
-            {"videos": videos, "full_path": full_path},
+            {
+                "videos": videos,
+                "full_path": full_path,
+                "count_videos": count_videos
+            },
         )
-    data_context["use_category"] = USER_VIDEO_CATEGORY
+
     data_context["videos"] = videos
+    data_context["count_videos"] = count_videos
+    data_context["types"] = request.GET.getlist("type")
+    data_context["owners"] = request.GET.getlist("owner")
+    data_context["disciplines"] = request.GET.getlist("discipline")
+    data_context["tags_slug"] = request.GET.getlist("tag")
+    data_context["cursus_selected"] = request.GET.getlist("cursus")
     data_context["full_path"] = full_path
+    data_context["cursus_list"] = CURSUS_CODES
+    data_context["use_category"] = USER_VIDEO_CATEGORY
     data_context["page_title"] = _("My videos")
 
     return render(request, "videos/my_videos.html", data_context)
 
 
-def get_videos_list(request):
+def get_videos_list():
```

Comment:
Pense à mettre des pydocs au format """Render the main list of videos."""

Reply:
doux jésus

Reply:
my bad j'ai fait les autres mais pas celui-ci